checks/provenance/check.go

```diff
@@ -37,54 +42,34 @@ func (p *check) Check(ctx context.Context, i voucher.ImageData) (bool, error) {
 		return false, fmt.Errorf("Got %d items for: %s", len(items), i.String())
 	}
 
-	item, ok := items[0].(*grafeas.Item)
-	if !ok {
-		return false, fmt.Errorf("response from MetadataClient is not an grafeas.Item")
-	}
-
-	buildDetails := item.Occurrence.GetBuild()
-	if validateProvenance(buildDetails) && validateArtifacts(i, buildDetails) {
-		log.Infof("Validated image provenance and artifacts for: %s", i.String())
+	buildDetail := items[0]
+	if ok, logerr := validateProvenance(p, buildDetail); ok && validateArtifacts(i, buildDetail) {
 		return true, nil
+	} else {
+		err = fmt.Errorf(logerr)
 	}
 
-	return false, nil
+	return false, err
 }
 
-func validateProvenance(details *build.Details) (trusted bool) {
-	// get trusted things
-	trustedBuilderIdentities := voucher.ToMapStringBool(viper.GetStringMap("trusted-builder-identities"))
-	trustedBuilderProjects := voucher.ToMapStringBool(viper.GetStringMap("trusted-projects"))
-
-	provenance, err := base64.StdEncoding.DecodeString(details.ProvenanceBytes)
-	if err != nil {
-		log.Errorf("Error decoding provenance: %v", err)
-		return
-	}
-
-	// unmarshal against provenance bytes
-	if err := json.Unmarshal(provenance, details); err != nil {
-		log.Errorf("Provenance bytes do not match details: %v", err)
-		return
-	}
-
-	if !trustedBuilderIdentities[details.Provenance.Creator] {
-		log.Errorf("Builder identity not trusted: %s", details.Provenance.Creator)
+func validateProvenance(p *check, detail voucher.BuildDetail) (trusted bool, logerr string) {
```

Comment:
Rather than returning a string here we should return a Error, the way you are doing on line 49:

```suggestion
func validateProvenance(p *check, detail voucher.BuildDetail) (trusted bool, err error) {
```

Reply:
Done.